### PR TITLE
[QPPA-920] Use react router (Part 1 of 3)

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**
Uses react router for routing to landing page, tutorials, developer preview and schemas.

[QPPA-920](https://jira.cms.gov/browse/QPPA-920)

Detail:
In preparation for adding a left nav bar to all pages, I create a [tech spec](https://docs.google.com/document/d/1CL-CXHcIJifPJYUkIQQaBN15nRuDl5vSC_f3CEJv0wc/edit#) which includes a refactoring of the navigation implementation so that it is simpler to extend to additional components. This is the first PR of 3 to implement the new navigation and styling on all pages.

Test plan:
* [x] adhoc testing via UI
* [x] updated unit tests

aug-16-2017 12-55-51

### CODE REVIEW

File: src/components/introduction.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          <p className="ds-text">Walk through how to create a new submission, submit measures and receive real-time scoring in the below tutorial.</p>
+          <Link to="/tutorial" className="ds-c-button ds-c-button--primary">Start the tutorial</Link>
+
+          <p><br/>Return to the <a href="https://qpp.cms.gov/resources/developers">QPP Developer Tools</a>.</p>
+        </div>
+      </section>
+  )}
+}
+
+export default Introduction;
```
Comment (kencheeto) [inline src/components/introduction.js]: 💯 for splitting this out

File: src/index.js [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+import { BrowserRouter } from 'react-router-dom'
 import App from './components/app';
 import './styles/index.css';
 
 ReactDOM.render(
-  <App url={document.URL}/>,
+  // According to the docs, forceRefresh is required for browsers which don't
+  // support pushState. However, even when this is set to false and 'pushState'
+  // in window.history is 'true', clicked links did not impact the content.
+  <BrowserRouter forceRefresh={true} basename="/qpp-submissions-docs/">
```
Comment (kencheeto) [inline src/index.js]: hm, this is worth digging more into I think. not sure if it's related, but clicking 'back' after navigating off of the root doesn't work either

File: src/components/app.test.js [hunk 3]
```diff
@@ -1,26 +1,34 @@
-import React from 'react';
-import {shallow, mount} from 'enzyme';
-import App from './app';
-import BasicTutorial from './tutorials/basic-tutorial';
-import DeveloperPreview from './developer-preview';
+import { render } from 'react-dom'
+import React from 'react'
+import { MemoryRouter } from 'react-router-dom'
```
Comment (kencheeto) [inline src/components/app.test.js]: why are we using MemoryRouter in tests?

File: src/components/app.jsx [hunk 4]
```diff
@@ -1,62 +1,56 @@
 import React from 'react';
-import url from 'url';
 import '../../node_modules/uswds/dist/js/uswds.js';
 import '../../node_modules/@cmsgov/design-system-core/dist/index.css';
+import { Route } from 'react-router-dom'
```
Comment (kencheeto) [inline src/components/app.jsx]: i think there are a lot of missing semicolons - i'll add our eslint rules to this project

File: src/components/app.test.js (hunk 3 above)
Comment (abarciauskas-bgse) [inline src/components/app.test.js]: That's the react router standard for tests: https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/guides/testing.md

Comment (kencheeto) [inline src/components/app.test.js]: oooo gtk! thanks

File: src/components/app.test.js [hunk 5]
```diff
@@ -1,26 +1,34 @@
...
+import App from './app'
 
-it('renders without crashing', () => {
-  shallow(<App url={document.URL}/>);
-});
```
Comment (kencheeto) [inline src/components/app.test.js]: i thought this test was helpful - i could see if App itself could render (and avoid trying to render nested components)

File: src/components/app.test.js [hunk 6]
```diff
@@ -1,26 +1,34 @@
...
+it('displays default content', () => {
+  const div = document.createElement('div')
+  render((
+    <MemoryRouter>
+      <App />
+    </MemoryRouter>
+  ), div)
```
Comment (kencheeto) [inline src/components/app.test.js]: thoughts about using enzyme's `mount` when we need deep rendering (like here) and `shallow` when we don't?

Comment (abarciauskas-bgse) [inline src/components/app.test.js]: @kencheeto tried mount and tests were failing, but I added back the `renders without crashing` test!

File: package.json [hunk 7]
```diff
@@ -35,7 +35,9 @@
     "react-scripts": "1.0.10",
     "react-tabs": "0.8.3",
     "react-test-renderer": "15.6.1",
-    "uswds": "1.0.0"
+    "uswds": "1.0.0",
+    "jest-cli": "^20.0.4",
```
Comment (kencheeto) [inline package.json]: i still don't think this is required - i removed it locally and the app still builds for me and tests pass

File: src/index.js [hunk 8]
```diff
@@ -1,9 +1,15 @@
...
 import ReactDOM from 'react-dom';
+import { BrowserRouter } from 'react-router-dom';
 import App from './components/app';
 import './styles/index.css';
 
 ReactDOM.render(
-  <App url={document.URL}/>,
+  // According to the docs, forceRefresh is required for browsers which don't
+  // support pushState. However, even when this is set to false and 'pushState'
+  // in window.history is 'true', clicked links did not impact the content.
```
Comment (kencheeto) [inline src/index.js]: we can remove these comments now!

**Pull request closed without merging**